lib/elastic/helpers.js

```diff
@@ -0,0 +1,297 @@
+  then: function (propertyName, ops) {
+    return ops;
+  }
+}];
+
+// Required for `_.listDeepObjects`
+_.mixin(require('lodash-ny-util'));
+
+/**
+ * A function which returns a function to help
+ * compare two values. Pass in the value you exect
+ * and then invoke the function with the value you're
+ * testing as the new argument.
+ *
+ * @param  {String} expected
+ * @return {Function}
+ */
+function compare(expected) {
+  return function (comparison) {
+    return expected === comparison;
+  };
+}
+
+/**
+ * @param {object} op
+ * @returns {object}
+ */
+function parseOpValue(op) {
+  try {
+    if (_.isString(op.value)) {
+      op.value = JSON.parse(op.value);
+    }
+  } catch (ex) {
+    log('warn', op, ex.stack);
+  }
+  return op;
+}
+
+/**
+ * Convert thing to string or [string] -- ES doesn't know the difference.
+ * @param {*} value
+ * @returns {string|[string]}
+ */
+function convertObjectToString(value) {
+  if (_.isArray(value)) {
+    value = _.map(value, function (property) {
+      var innerValue;
+
+      if (_.isPlainObject(property)) {
+        innerValue = _.find(_.pickBy(property, _.isString));
+      } else if (_.isString(property)) {
+        innerValue = property;
+      } else {
+        log('warn', 'Bad String or [String] type', value);
+      }
+
+      return innerValue;
+    });
+  } else if (_.isPlainObject(value)) {
+
+    // Special special!  An array of 'items' means this object is actually an array with properties (the array's items
+    // are moved to the 'items' property
+    if (_.isArray(value.items)) {
+      value = convertObjectToString(value.items);
+    } else {
+      // take first property of the object that is a string
+      value = _.find(_.pickBy(value, _.isString));
+    }
+
+  } else if (!_.isString(value)) {
+    log('warn', 'Bad String or [String] type', value);
+    value = null;
+  }
+
+  return value;
+}
+
+/**
+ * If any of the properties that should be
+ * strings are references, resolve them.
+ *
+ * @param {string} propertyName
+ * @returns {Function}
+ */
+function resolveReferencesForPropertyOfStringType(propertyName) {
+  return function (ops) {
+    return bluebird.all(_.map(ops, function (op) {
+      var result, value = op.value[propertyName];
+
+      if (value !== null && value !== undefined) {
+        if (value[refProp]) {
+          result = db.get(value[refProp]).then(JSON.parse).then(function (referencedValue) {
+            op.value[propertyName] = convertObjectToString(referencedValue);
+            return op;
+          });
+        } else {
+          op.value[propertyName] = convertObjectToString(value);
+          result = op;
+        }
+      }
+
+      return result;
+    })).then(_.compact);
+  };
+}
+
+/**
+ * Storing null and undefined for a mapped property
+ * confuses ES (doesn't error, but creates odd behavior).
+ *
+ * @param {string} propertyName
+ * @returns {Function}
+ */
+function removeEmptyValuesForProperty(propertyName) {
```

Comment:
Does this get used anywhere? Popped into my head when I saw this: https://github.com/nymag/amphora/pull/339/files#diff-6612bbbf33f2f172a7fe9e466fbfff28R214